### ISSUE TITLE
Automatically build some `*.dev` packages nightly

### ISSUE THREAD

**Issue by RalfJung** [coq/opam-coq-archive#246]
We should look into Travis' cron jobs to automatically build some of our `*.dev` packages every night, so that we can detect build failure incurred from Coq changes.

**Comment by ejgallego** [comment]
I am not against this but it seems to me a bit heavy-handed. Why not to remove the source of de-synchronization in the first place?

We could also add a packaging test to the main Coq jobs. I hope we will have soon a large-enough CI power.

**Comment by Zimmi48** [comment]
> We could also add a packaging test to the main Coq jobs. I hope we will have soon a large-enough CI power.

We could but I'm unconvinced that we are actually going to scale up our CI power so distributing the builds across repository looks like a better approach in the meantime.

**Comment by RalfJung** [comment]
> Why not to remove the source of de-synchronization in the first place?

So, you mean, use `make uninstall`? (a) I think that would be great, but maybe there are reasons for why it was not used so far? (b) This only makes sense if the `uninstall` target is actually tested. In particular once/if we have #247, a CI cron job would also test that `make uninstall` does not forget to uninstall something.

**Comment by ejgallego** [comment]
IMHO the `make uninstall` coq target should be tested in Coq itself. @JasonGross has indeed made the point in the past that we don't properly test `install/uninstall` and he's 200% spot on.

In the medium term, I hope that commands such `install/uninstall` are an automated byproduct of the build system so they don't have to be maintained anymore.

**Comment by clarus** [comment]
@RalfJung I do not know if it solves your problem, but the opam bench https://coq-bench.github.io/ now tests (again) the development packages (results in the `Development` column).

**Comment by RalfJung** [comment]
@clarus interesting! However, I don't understand this introduction:

> We bench the Coq OPAM packages, not Coq itself or the contribs. The compilation times are only informative.

So, if this is not benching building Coq, and not benching building stuff with Coq, then what is it benching...? The first part makes it sound like `time opam install coq`, but then it says "not Coq itself" which negates that?

**Comment by palmskog** [comment]
First, big thanks to @clarus for setting this up!

I'm sure @clarus can confirm, but from looking at the results, it seems clear to me that by "contribs", he's referring to the legacy `coq-contribs` projects. So what is reported is as far as I can see `time opam install coq-<pkg>`, with the precondition that `opam install coq.<version>` has been run first.

Some quick feedback that could make this even more useful: `8.7.dev` is not really an interesting Coq version to test, I would much rather see `8.10.dev` and even `dev`, which a lot of packages in `extra-dev` actually target. Also, I'm not sure why non-extra-dev package versions (regular releases like `1.0.5` and so on) are reported in the table for `extra-dev`? To me this is more like: "testing development Coq versions with `released` packages.

**Comment by clarus** [comment]
Yes, this was referring to the "contribs" of the `coq-contribs` project, at the time there was no opam and the contribs were the primary way to publish Coq projects.

OK for removing old development versions of Coq from the bench. To me testing regular releases of packages with development versions of Coq could be useful to check if there are not (too much) regressions on existing packages.
